src/buffer/view.rs

```diff
@@ -658,15 +658,16 @@ impl<'a, T> BufferSlice<'a, [T]> where [T]: Content + 'a {
     /// This method builds an object that represents a slice of the buffer. No actual operation
     /// OpenGL is performed.
     #[inline]
-    pub fn slice(&self, range: Range<usize>) -> Option<BufferSlice<'a, [T]>> {
-        if range.start > self.len() || range.end > self.len() {
+    pub fn slice<R: RangeArgument<usize>>(&self, range: R) -> Option<BufferSlice<'a, [T]>> {
+		let clos = |val: &usize| { *val };
+        if range.start().map_or(0, &clos) > self.len() || range.end().map_or(0, &clos) > self.len() {
             return None;
         }
 
         Some(BufferSlice {
             alloc: self.alloc,
-            bytes_start: self.bytes_start + range.start * mem::size_of::<T>(),
-            bytes_end: self.bytes_start + range.end * mem::size_of::<T>(),
+            bytes_start: self.bytes_start + range.start().map_or(0, &clos) * mem::size_of::<T>(),
```

Comment:
Anyway, I'd prefer if you just repeated the closure instead of defining it at the top.
Just something like `.map_or(0, |&v| v)`.